src/analysis/ReachingDefinitions/Srg/AssignmentFinder.h

```diff
@@ -0,0 +1,124 @@
+#ifndef _DG_ASSIGNMENTFINDER_H_
+#define _DG_ASSIGNMENTFINDER_H_
+
+#include <unordered_map>
+#include <vector>
+
+#include "analysis/ReachingDefinitions/ReachingDefinitions.h"
+#include "analysis/SubgraphNode.h"
+
+namespace dg {
+namespace analysis {
+namespace rd {
+namespace srg {
+
+using AssignmentMap = std::unordered_map<dg::analysis::rd::RDNode *, std::vector<dg::analysis::rd::RDNode *>>;
+
+/**
+ * Constructs Def->Use graph from RDNode-s
+ */
+class AssignmentFinder
+{
+private:
+    unsigned DFS = 10;
+    using NodeT = dg::analysis::rd::RDNode;
+
+    /**
+     * Returns all nodes from graph with given root in BFS order
+     */
+    std::vector<NodeT *> bfs(NodeT *root)
+    {
+        assert(root && "need root");
+        std::vector<NodeT *> result;
+
+        unsigned dfsnum = DFS++;
+
+        ADT::QueueFIFO<NodeT *> fifo;
+        fifo.push(root);
+        root->dfsid = dfsnum;
+
+        while (!fifo.empty()) {
+            NodeT *cur = fifo.pop();
+            result.push_back(cur);
+
+            for (NodeT *succ : cur->successors) {
+                if (succ->dfsid != dfsnum) {
+                    succ->dfsid = dfsnum;
+                    fifo.push(succ);
+                }
+            }
+        }
+
+        return result;
+    }
+
+    bool isDefinition(NodeT *node)
```

Comment:
The name of this method is misleading.

Reply:
You're right, that isn't how the method should be called. I'll try to come up with something better.

Reply:
I think that simple `isAllocation` would do :)